src/DefaultManifold.jl

```diff
@@ -8,12 +8,31 @@ to build one's own manifold. It is a simplified/shortened variant of `Euclidean`
 This manifold further illustrates how to type your manifold points and tangent vectors. Note
 that the interface does not require this, but it might be handy in debugging and educative
 situations to verify correctness of involved variabes.
+
+# Constructor
+
+    DefaultManifold(n::Int...; field = ℝ, parameter::Symbol = :field)
+
+
+Arguments:
+
+- `n`: shape of array representing points on the manifold.
+- `field`: field over which the manifold is defined. Either `ℝ`, `ℂ` or `ℍ`.
+- `parameter`: whether a type parameter should be used to store `n`. By default size
+  is stored in a field. Value can either be `:field` or `:type`.
 """
-struct DefaultManifold{𝔽,T<:NTuple{N,Int} where {N}} <: AbstractManifold{𝔽}
+struct DefaultManifold{𝔽,T<:AbstractManifoldParameter} <: AbstractManifold{𝔽}
     size::T
 end
-function DefaultManifold(n::Vararg{Int}; field = ℝ)
-    return DefaultManifold{field,typeof(n)}(n)
+function DefaultManifold(n::Vararg{Int}; field = ℝ, parameter::Symbol = :field)
+    if parameter === :field
+        size = FieldParameter(n)
```

Comment:
one could even just use a number or an attar here, right?

Though then the  `get_parameter` will be a bit more difficult (and the current constraint on `T` has to be let go of).

Reply:
Hm, technically we could have
```julia
get_parameter(::TypeParameter{T}) where {T} = T
get_parameter(P) = P
```
i.e. everything other than `TypeParameter` would be stored without any wrapper. That would also be fine.

Reply:
Yes, something like that. Do you think that might be confusing? I think that is an easier solution and even better `size` would really be an int or array again (for field type) :)

Reply:
I've implemented your idea and it works well, thanks :slightly_smiling_face: . I like it.

Reply:
Nice! I like the result as well – and it was more by accident that I saw that idea.